### ISSUE TITLE
Add Random Elastic Transform

### ISSUE THREAD

**Issue by innat** [innat/medic-ai#105]
**Is your feature request related to a problem? Please describe.**

Elastic transform or deformation is widely used in medical image, i.e. MRI, CT. Also in nnUNet pipeline.

**Describe the solution you'd like**

A tensorflow based implementation, fast execution. Design notes:

  1. **Gaussian smoothing**  -> separable depthwise conv (`ops.conv`)
  2. **map_coordinates**    -> batched N-linear interpolation via a flat gather
  3. masks are warped with NEAREST-neighbor, not N-linear + rounding, to
     avoid fabricating phantom classes at multi-class label boundaries.

We have built a prototype and so far looks fine in terms of quality and speed.

```python
import medicai.transforms import RandomElasticTransform

transform = RandomElasticTransform(
     alpha=15.0, sigma=4.0, seed=0
)
```

<img width="698" height="357" alt="Image" src="https://github.com/user-attachments/assets/46060ca4-b80e-4278-bb27-df980cf50f0c" />

```python
--- 3D volume, batch=4  shape=(4, 64, 64, 64, 1)  backend=tensorflow ---
  eager:     174.89 ms  (+/-3.34, best 170.52)
  compiled:    7.71 ms  (+/-0.17, best 7.50)

--- 2D image, batch=8  shape=(8, 128, 128, 1)  backend=tensorflow ---
  eager:      53.61 ms  (+/-0.82, best 51.87)
  compiled:    1.35 ms  (+/-0.06, best 1.25)
```

**Describe alternatives you've considered**

To use Keras official [implementation](https://keras.io/api/layers/preprocessing_layers/image_augmentation/random_elastic_transform/). However, few issue with it:

1. It doesn't support 3D. We need shape agnostic, should work on 2D and 3D.
2. It is implemented with `keras.ops`, that makes it backend agnostic but also problematic for jax and torch backend if we include it to `tf.data` API.
3. Last but not least, super slow, in both CPU and GPU model. More details: https://github.com/keras-team/keras/issues/23245#issue-4864460293

**Additional context**

- Implementation should be faithfull and fast enough.
